src/vm/readytoruninfo.cpp

```diff
@@ -648,12 +648,40 @@ PCODE ReadyToRunInfo::GetEntryPoint(MethodDesc * pMD, BOOL fFixups /*=TRUE*/)
             offset -= val;
         }
 
+#ifndef CROSSGEN_COMPILE
+#ifdef PROFILING_SUPPORTED
+        BOOL fShouldSearchCache = TRUE;
+        {
+            BEGIN_PIN_PROFILER(CORProfilerTrackCacheSearches());
+            g_profControlBlock.pProfInterface->
+                JITCachedFunctionSearchStarted((FunctionID)pMD, &fShouldSearchCache);
+            END_PIN_PROFILER();
+        }
+        if (!fShouldSearchCache)
+        {
+            return NULL;
+        }
+#endif // PROFILING_SUPPORTED
+#endif // CROSSGEN_COMPILE
+
         if (fFixups)
         {
             if (!m_pModule->FixupDelayList(dac_cast<TADDR>(m_pLayout->GetBase()) + offset))
                 return NULL;
         }
 
+#ifndef CROSSGEN_COMPILE
+#ifdef PROFILING_SUPPORTED
+        if (fShouldSearchCache)
```

Comment:
This line is unnecessary - you can only get here with `fShouldSearchCache == TRUE`